phac-nml/irida-next
# Member create modal

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds functionality to add a member to a group/project using a modal and to use turbo_stream.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/75b133f9-4d80-45a0-a71c-3cba8a6d1622)

![image](https://github.com/phac-nml/irida-next/assets/25466211/ea94dbdd-3dcd-4d09-95c8-a61e551c861a)

![image](https://github.com/phac-nml/irida-next/assets/25466211/cb4877d6-cf1d-408b-a6ac-e987d4bc4e1a)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as any user
2. Visit a groups members page
3. Click the `Add New Member` button
4. A modal should pop up for you to select a user and their access level
5. Select a user and their access level
6. Click the `Add member to group` button
7. The table and pagination should update to include this user
8. Verify in the network tab of the browser dev tools that the request to add a member to a group is sent via turbo_stream
9. Verify the same functionality for a project
10. Verify in the network tab of the browser dev tools that the request to add a member to a project is sent via turbo_stream

Note: If trying to add a member to a subgroup or a project where the user also has membership in one of it's ancestors, with a lower access level you should see an error in the modal such as:

![image](https://github.com/phac-nml/irida-next/assets/25466211/0d6a2d2f-a48c-47df-b981-aa923bab2c69)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
